gtsfm/utils/metrics.py

```diff
@@ -0,0 +1,197 @@
+"""Utilities to compute and save evaluation metrics.
+
+Authors: Akshay Krishnan
+"""
+import json
+import numpy as np
+import os
+from dask.delayed import Delayed
+from typing import Dict, List, Optional, Tuple, Union
+
+from gtsam import (Point3, Rot3, Pose3, Unit3)
+
+import gtsfm.utils.geometry_comparisons as comp_utils
+from gtsfm.common.sfm_result import SfmResult
+
+# A StatsDict is a dict from string to optional floats or their lists.
+StatsDict = Dict[str, Union[Optional[float], List[Optional[float]]]]
+
+
+def get_errors_statistics(errors: List) -> StatsDict:
+    """Computes statistics (min, max, median) on the given list of errors
+
+    Args:
+        errors: List of errors for a metric.
+
+    Returns: 
+        A dict with keys min_error, max_error, median_error, 
+        and errors_list mapping to the respective stats.
+    """
+
+    metrics = {}
+    valid_errors = [error for error in errors if error is not None]
+    metrics["median_error"] = np.median(valid_errors)
+    metrics["min_error"] = np.min(valid_errors)
+    metrics["max_error"] = np.max(valid_errors)
+    metrics["errors_list"] = errors
+    return metrics
+
+
+def compute_rotation_angle_metrics(
+        wRi_list: List[Optional[Rot3]],
+        gt_wRi_list: List[Optional[Pose3]]) -> StatsDict:
+    """Computes statistics for the angle between estimated and GT rotations. 
+
+    Assumes that the estimated and GT rotations have been aligned and do not 
+    have a gauge freedom. 
+
+    Args: 
+        wRi_list: List of estimated camera rotations. 
+        gt_wRi_list: List of ground truth camera rotations. 
+
+    Returns: 
+        A statistics dict of the metrics errors in degrees. 
+    """
+    errors = []
+    for (wRi, gt_wRi) in zip(wRi_list, gt_wRi_list):
+        angle = comp_utils.compute_relative_rotation_angle(wRi, gt_wRi)
+        errors.append(angle)
+    return get_errors_statistics(errors)
+
+
+def compute_translation_distance_metrics(
```

Comment:
unit test here would be nice too

Reply:
Do we really need unit tests for this file? All the logic has been moved to geometry_comparisons utils and is unit tested. Here we are using the same logic on a container and logging to JSON.

Reply:
i'm a bit ambivalent, either way